utils/defaulter_test.go

```diff
@@ -572,3 +573,160 @@ func TestGetDefaulter_Konnect(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckRestrictedFields(t *testing.T) {
+	assert := assert.New(t)
+
+	testCases := []struct {
+		desc             string
+		entity           *kong.Service
+		restrictedFields []string
+		invalidFields    []string
+	}{
+		{
+			desc: "no restricted fields",
+			entity: &kong.Service{
+				ID:   kong.String("testID"),
+				Name: kong.String("testName"),
+			},
+			restrictedFields: []string{},
+		},
+		{
+			desc: "one restricted fields",
+			entity: &kong.Service{
+				ID:   kong.String("testID"),
+				Name: kong.String("testName"),
+			},
+			restrictedFields: []string{"ID"},
+			invalidFields:    []string{"ID"},
+		},
+		{
+			desc: "multiple restricted fields",
+			entity: &kong.Service{
+				ID:   kong.String("testID"),
+				Name: kong.String("testName"),
+				Port: kong.Int(80),
+			},
+			restrictedFields: []string{"ID", "Name", "Port"},
+			invalidFields:    []string{"ID", "Name", "Port"},
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			err := checkEntityDefaults(tC.entity, tC.restrictedFields)
+			if len(tC.invalidFields) > 0 {
```

Comment:
What about having two more fields in the testcase struct like these:

```
		wantErr          bool
		expectedErr      string
```

and refactor this like:

```
if (err != nil) != tC.wantErr {
	t.Errorf("got error = %v, expected error = %v", err, tC.wantErr)
}
if tC.expectedErr != "" {
	assert.Equal(err.Error(), tC.expectedErr)
}
```

This makes you test the whole error string

Reply:
I agree! fixing this.